### ISSUE TITLE
PaletteReference Lint check doesn't handle player palettes properly

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#9343]
...when they are enabled via boolean.
Example: the D2k deviator trail palette (#9340).
It gives a lint error even though there is nothing wrong with the yaml and everything works fine in-game.

The only reason it didn't error before was that the deviator used `Bullet` projectile, which lacks a `[PaletteReference]` for the `TrailPalette`.

**Comment by reaperrr** [comment]
Note: Until this is fixed, Travis will always fail once #9340 is merged or `[PaletteReference]` added to `Bullet`'s `TrailPalette` or `CreateEffectWarhead`'s `ExplosionPalette`.

**Comment by pchote** [comment]
I complained about this extensively around the time of #8749, and even wrote some code showing how to do it.  My objections were overruled in favor of filing #9101 instead.

**Comment by Mailaender** [comment]
That would have added a lot of complexity as it would have required upgrade rules, because we handle player palette enablement inconsistently around the engine. I was already having trouble finding and fixing the existing palette references across 4 mods.

**Comment by Mailaender** [comment]
Duplicate of https://github.com/OpenRA/OpenRA/issues/9101.
